### PR TITLE
feat(NODE-5409)!: allow socks to be installed optionally

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- Add getSocks method to deps
- update dependencies test for socks, and general imports

##### Is there new documentation needed for these changes?

No

#### What is the motivation for this change?

See highlight

### Allow socks to be installed optionally

The driver uses the `socks` dependency to connect to `mongod` or `mongos` through a [SOCKS5 proxy](https://en.wikipedia.org/wiki/SOCKS).  `socks` used to be a required dependency of the driver and was installed automatically.  Now, `socks` is a `peerDependency` that must be installed to enable `socks` proxy support. 

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: src/client-side-encryption/stateMachine.js [hunk 1]
```diff
@@ -4,12 +4,25 @@ import * as tls from 'tls';
...
 import { MongoNetworkTimeoutError } from '../error';
 import { debug, databaseNamespace, collectionNamespace } from './common';
 import { MongoCryptError } from './errors';
 import { BufferPool } from './buffer_pool';
 import { serialize, deserialize } from '../bson';
+import { getSocks } from '../deps';
+
+/** @type {import('../deps').SocksLib | null} */
+let socks = null;
+function loadSocks() {
```
Comment (baileympearson) [inline src/client-side-encryption/stateMachine.js:16]: Why is this necessary?  `makeModuleError` returns a proxy that throws whenever a key that isn't `kModuleError` is accessed.  So I'd expect we would just do the following (where it's used):
```typescript
const socks = loadSocks();
await socks.SocksClient(...)
```

File: src/cmap/connect.ts [hunk 2]
```diff
@@ -419,6 +419,18 @@ function makeConnection(options: MakeConnectionOptions, _callback: Callback<Stre
   }
 }
 
+let socks: SocksLib | null = null;
+function loadSocks() {
+  if (socks == null) {
+    const socksImport = getSocks();
```
Comment (baileympearson) [inline src/cmap/connect.ts:425]: same comment as in stateMachine.js.

**Review by baileympearson — CHANGES_REQUESTED** [review]
questions about how we're importing socks

File: src/client-side-encryption/stateMachine.js (hunk 1 above)
Comment (nbbeeken) [reply src/client-side-encryption/stateMachine.js:16]: I'm alinging with the approach taken for kerberos, zstd, snappy. The goal is to not invoke the import operation more than once, determined by the nullishness of the `socks` variable.

File: src/cmap/connect.ts (hunk 2 above)
Comment (nbbeeken) [reply src/cmap/connect.ts:425]: ditto response

**Review by baileympearson — APPROVED** [review]
